lisp/ess-r-package.el

```diff
@@ -363,6 +364,12 @@ With prefix argument ARG, run tests on current file only."
 })
 ")
 
+(defun ess-r-package-save-buffers ()
+  "Offer to save modified files in the current project.
+Respects `ess-silently-save', which see."
+  (dolist (file (project-files (project-current)))
```

Comment:
What does project-files do internally? If it finds all project files then this logic is sub-optimal. What we really need is to check only open files. Those are likely a handful and already have our R project root cached. So mapping through buffer list is likely a much, much faster alternative than this, also backward compatible from the start.

Reply:
> then this logic is sub-optimal. 

Well, perhaps. If you've got a ton of buffers open but only a few files in your project, then this way is faster. But since `project-files` doesn't exist yet, I went with your idea anyway :-)